src/QuicrDelegates.cpp

```diff
@@ -109,7 +131,47 @@ void QuicrTransportSubDelegate::onSubscribedObject(const quicr::Name& quicrName,
     currentGroupId = groupId;
     currentObjectId = objectId;
 
-    qDelegate->subscribedObject(std::move(data), groupId, objectId);
+    // Decrypt the received data using sframe
+    try
+    {
+        auto key_id_length = QUICVarIntSize(data.data());
+        if (data.size() <= key_id_length)
+        {
+            logger.log(qtransport::LogLevel::error,
+                       "Received an object with a corrupt key ID");
+            return;
+        }
+        auto sframe_key_id = QUICVarIntDecode(data.data());
+        if (sframe_key_id == std::numeric_limits<std::uint64_t>::max())
+        {
+            logger.log(qtransport::LogLevel::error,
+                       "Received an object with an invalid key ID");
+            return;
+        }
+        quicr::bytes output_buffer(data.size() - key_id_length);
+        auto cleartext = sframe_context.unprotect(
+            sframe_key_id,
+            quicr::Namespace(quicrNamespace.name(), Quicr_SFrame_Sig_Bits),
```

Comment:
```suggestion
            quicr::Namespace(quicrName, Quicr_SFrame_Sig_Bits),
```

Basically, I think you just need to ignore the namespace that the object tracks, and just extract the fixed bits from the object name you're given here.

Reply:
And likewise on encrypt.